clients/src/main/java/org/apache/kafka/clients/consumer/internals/CommitRequestManager.java

```diff
@@ -210,17 +210,19 @@ public CompletableFuture<Void> maybeAutoCommitAllConsumed() {
         return maybeAutoCommit(subscriptions.allConsumed());
     }
 
+    boolean canAutoCommit() {
```

Comment:
Unhappy with the naming, but can't seem to find a better way to restructure this at the moment.  Didn't like it because there is an infinite variation of canXxxx().